src/node/desktop/src/main/main-window.ts

```diff
@@ -217,8 +217,12 @@ export class MainWindow extends GwtWindow {
 
     this.executeJavaScript('window.desktopHooks.getActiveProjectDir()')
       .then((projectDir) => {
+        if (process.platform == 'darwin') app.dock.setBadge('');
+
         if (projectDir.length > 0) {
           this.window.setTitle(`${projectDir} - ${appState().activation().editionName()}`);
+       
+          if (process.platform == 'darwin') app.dock.setBadge(projectDir);
```

Comment:
How do longer project names appear here? Are they truncated sensibly?

Reply:
No, they are truncated on the middle automatically by Electron in this pattern: `[first 4 letters]...[last 5 letters]` . The QT version shows the first 14 letters . I could show only the first X amount of letters instead of letting Electron clip it automatically... How should this be done? 

![Screen Shot 2022-06-26 at 23 20 08](https://user-images.githubusercontent.com/36674530/175848210-d039470f-c10d-4025-b7f9-1506443dcd48.png)

![Screen Shot 2022-06-26 at 23 19 30](https://user-images.githubusercontent.com/36674530/175848217-7d92a594-f535-4720-bb65-80b1f48ec2d7.png)